packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -34,9 +34,7 @@ export type OptionsSpec<TOptions extends Options> = {
   }
 }
 
-export type MetricsV2 = {
-  [key: string]: Metric
-}
+export type MetricsV2 = Record<any, Metric>
```

Comment:
We could be nits here with 
export type MetricsV2<TMetricKeys extends string> = Record<TMetricKeys, Metrics>
But that adds little to no value while bubbling up generic typescript types making base service more complicated.